src/Parser/Visitor/DocBlockTypeAnnotations.php

```diff
@@ -8,21 +8,24 @@
 use phpDocumentor\Reflection\Types\Context;
 use phpDocumentor\Reflection\Types\Object_;
 use PhpParser\Node;
-use ReflectionMethod;
 
 class DocBlockTypeAnnotations extends NamespaceCollectingVisitor
 {
-
     /** @var string */
-    private $lastClass;
+    private $lastNamespace;
+
+    /** @var string[] */
+    private $useStatements = [];
 
     public function enterNode(Node $node)
     {
-        if ($node instanceof Node\Stmt\ClassLike && $node->namespacedName !== null) {
-            $this->lastClass = $node->namespacedName->toString();
-        } elseif ($node instanceof Node\Stmt\ClassMethod && $this->lastClass) {
-            $contextFactory = new \phpDocumentor\Reflection\Types\ContextFactory();
-            $context = $contextFactory->createFromReflector(new ReflectionMethod($this->lastClass, (string) $node->name));
+        if ($node instanceof Node\Stmt\Namespace_) {
+            $this->lastNamespace = $node->name->toString();
```

Comment:
Heads up: `$node->name` can either be `PhpParser\Node\Name` or `null` - please add a null check before calling `toString`.

Reply:
Fixed